crates/blob-store/src/implementors/azblob.rs

```diff
@@ -0,0 +1,216 @@
+#[derive(Debug, Clone)]
+pub struct AzBlobContainer {
+    client: ContainerClient,
+}
+
+#[derive(Debug)]
+pub struct AzBlobReadStream {
+    blob_client: BlobClient,
+}
+
+#[derive(Debug, Clone)]
+pub struct AzBlobWriteStream {
+    client: BlobClient,
+}
+
+impl AzBlobContainer {
+    pub async fn new(slight_state: &BasicState, name: &str) -> Result<Self> {
+        let storage_account_name = get_from_state("AZURE_STORAGE_ACCOUNT", slight_state)
+            .await
+            .unwrap();
+        let storage_account_key = get_from_state("AZURE_STORAGE_KEY", slight_state)
+            .await
+            .unwrap();
+
+        let storage_credentials =
+            StorageCredentials::Key(storage_account_name.clone(), storage_account_key);
+        let service_client = BlobServiceClient::new(storage_account_name, storage_credentials);
+
+        let container_client = service_client.container_client(name);
+        if container_client.exists().await? {
+            Ok(Self {
+                client: container_client,
+            })
+        } else {
+            bail!(format!("container {name} not found"))
+        }
+    }
+}
+
+#[async_trait]
+impl ContainerImplementor for AzBlobContainer {
+    async fn name(&self) -> Result<String> {
+        Ok(self.client.container_name().to_owned())
+    }
+    async fn info(&self) -> Result<ContainerMetadata> {
+        let properties = self.client.get_properties().await?;
+        Ok(properties.container.into())
+    }
+    async fn list_objects(&self) -> Result<Vec<ObjectNameResult>> {
+        let mut stream = self.client.list_blobs().into_stream();
+        let mut results = vec![];
+        while let Some(value) = stream.next().await {
+            let value = value?;
+            results.push(value);
+        }
+
+        let mut result = vec![];
+        for list_blob in results {
+            for blob in list_blob.blobs.items {
+                let blob_name = match blob {
+                    BlobItem::Blob(b) => b.name.clone(),
+                    BlobItem::BlobPrefix(b) => b.name.clone(),
+                };
+                result.push(blob_name)
+            }
+        }
+        Ok(result)
+    }
+    async fn delete_object(&self, name: ObjectNameParam<'_>) -> Result<()> {
+        self.client
+            .blob_client(name)
+            .delete()
+            .delete_snapshots_method(DeleteSnapshotsMethod::Include)
+            .into_future()
+            .await?;
+        Ok(())
+    }
+    async fn delete_objects(&self, _names: Vec<ObjectNameParam<'_>>) -> Result<()> {
+        // TODO: there isn't an API in azure blob storage to do this directly
+        // if we are going to delete a lot of objects, we should use a batch delete
+        // otherwise, we run into issues with deleting half the objects and then
+        // failing
+        //
+        //
+        // followed up on https://github.com/Azure/azure-sdk-for-rust/issues/1249
+        todo!()
```

Comment:
@devigned I am thinking that the `delete-objects` API from wasi-blob-store is a bit too specific and demanding as that it is not something every blog store would provide. For example, there isn't delete-objects API in azure blob storage rust SDK, albert azure provides batch operations for deletions. 

I am in favor of removing this API form the spec. WDYT?

Reply:
Will keep it there